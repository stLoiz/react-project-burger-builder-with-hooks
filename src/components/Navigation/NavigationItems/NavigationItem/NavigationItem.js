import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';

import classes from './NavigationItem.module.css';

const NavigationItem = ({ children, link, exact }) => (
  <li className={classes.NavigationItem}>
    <NavLink to={link} activeClassName={classes.active} exact={exact}>
      {children}
    </NavLink>
  </li>
);

NavigationItem.defaultProps = {
  exact: false,
};
NavigationItem.propTypes = {
  children: PropTypes.node.isRequired,
  exact: PropTypes.bool,
  link: PropTypes.string.isRequired,
};
export default NavigationItem;
