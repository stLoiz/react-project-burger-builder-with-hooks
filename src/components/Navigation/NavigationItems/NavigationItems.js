import PropTypes from 'prop-types';
import React from 'react';

import classes from './NavigationItems.module.css';
import NavigationItem from './NavigationItem/NavigationItem';

const navigationItems = ({ isAuthenticated }) => (
  <ul className={classes.NavigationItems}>
    <NavigationItem link="/" exact>
      Burger Builder
    </NavigationItem>

    {isAuthenticated ? (
      <NavigationItem link="/orders">Orders</NavigationItem>
    ) : null}
    {isAuthenticated ? (
      <NavigationItem link="/logout">Logout</NavigationItem>
    ) : (
      <NavigationItem link="/auth">Sign in/up</NavigationItem>
    )}
  </ul>
);

navigationItems.propTypes = {
  /*
   * bool showing if the user is logged in or not
   */
  isAuthenticated: PropTypes.bool.isRequired,
};

export default navigationItems;
