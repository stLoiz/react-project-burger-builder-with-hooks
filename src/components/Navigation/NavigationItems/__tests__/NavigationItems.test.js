import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavigationItems from '../NavigationItems';
import NavigationItem from '../NavigationItem/NavigationItem';

describe('<NavigationItems/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavigationItems isAuthenticated />);
  });
  it('should match default snapshot on render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render two <NavigationItem/> elements if user is not authenticated', () => {
    wrapper.setProps({ isAuthenticated: false });
    expect(wrapper.find(NavigationItem)).toHaveLength(2);
  });
  it('should render three <NavigationItem/> elements if user is authenticated', () => {
    expect(wrapper.find(NavigationItem)).toHaveLength(3);
  });
  it('should render the Logout <NavigationItem/> element if user is authenticated', () => {
    expect(
      wrapper.contains(<NavigationItem link="/logout">Logout</NavigationItem>),
    ).toEqual(true);
  });
});
