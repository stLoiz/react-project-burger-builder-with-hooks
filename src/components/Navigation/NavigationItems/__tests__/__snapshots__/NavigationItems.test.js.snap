// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems/> should match default snapshot on render 1`] = `
<ul
  className="NavigationItems"
>
  <navigationItem
    exact={true}
    link="/"
  >
    Burger Builder
  </navigationItem>
  <navigationItem
    link="/orders"
  >
    Orders
  </navigationItem>
  <navigationItem
    link="/logout"
  >
    Logout
  </navigationItem>
</ul>
`;
